codex-ui/src/vue/components/button/Button.vue

```diff
@@ -1,32 +1,42 @@
 <template>
   <button
-    :class="[$style.button, `${$style.button}--${size}`, `${$style.button}--${style}`, 'text-ui-base-medium', icon && `${$style.button}--with-icon`]"
+    :class="[
+      $style.button,
+      `${$style.button}--${style}`,
+      iconType !== 'none' && `${$style.button}--icon-${iconType}`,
+
+      'text-ui-base-medium',
+    ]"
     :theme-accent="style === 'destructive' ? 'red' : undefined"
   >
     <Icon
       v-if="icon"
       :name="icon"
     />
-    <slot v-else />
+    <slot />
+    <Icon
+      v-if="trailingIcon"
+      :name="trailingIcon"
+    />
   </button>
 </template>
 
 <script setup lang="ts">
-import { computed, defineProps } from 'vue';
-import type { ButtonSize, ButtonStyle } from './Button.types';
+import { computed, defineProps, useSlots } from 'vue';
 import Icon from '../icon/Icon.vue';
 
 const props = withDefaults(
   defineProps<{
     /**
-     * The size of the button
+     * Button is primary by default
+     * Add the secondary prop to make it secondary
      */
-    size?: ButtonSize;
+    secondary?: boolean;
 
     /**
-     * The style of the button
+     * Pass this attribue for negative actions
      */
-    style?: ButtonStyle;
+    destructive?: boolean;
```

Comment:
maybe we can just rename style to smth else to avoid conflict? 
now if we specify both secondary and destructive will be applied only secondary style

Reply:
I think it's still more elegant way to write

```vue
<Button secondary />
```

instead of 

```vue
<Button style="secondary" />
```

and it is similar with native attributes of the [HTMLButtonElement](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button)